### PR TITLE
Fix path equivalences

### PR DESCRIPTION
**Pull request by aadcg**
This closes PR #45 and issue #59.

@Ambrevar

### CODE REVIEW

**Comment by Ambrevar** [comment]
Sorry for the delay, this fell under my radar.

Could you split the commit d9f93d1355ffe24935e4683913ffc5b6f3a6e36a so that the actual fix is not mixed together with the minor refactoring?

File: t/quri.lisp [hunk 1]
```diff
@@ -7,27 +7,51 @@
...
+        (make-uri :scheme "http" :host "test" :path "/" :fragment "bar")
+        "Differ by fragment.")
+  (is (uri "http://test")
+      (make-uri :scheme "http" :host "test" :path "")
+      "The NIL and empty string path are equivalent."))
+
 (subtest "uri="
-  (let ((prove:*default-test-function* #'uri-equal))
-    (is (uri "http://b.hatena.ne.jp")
-        (uri "http://b.hatena.ne.jp:80"))
```
Comment (Ambrevar) [inline t/quri.lisp:13]: You are no longer testing this.

Comment (Ambrevar) [reply t/quri.lisp:13]: You are also no longer testing with a URL that has seperate TLD, domain _and_ host.
I suggest you revert back to the original URL.  If you think we should test against a URL like `http://test`, it's OK to add it I think, but it does not replace the original test.

**Comment by aadcg** [comment]
> Could you split the commit d9f93d1 so that the actual fix is not mixed together with the minor refactoring?

Sure, done!

File: t/quri.lisp (hunk 1 above)
Comment (aadcg) [reply t/quri.lisp:13]: Good point. I'll address it.

**Comment by aadcg** [comment]
I share that concern as well. To test these equivalences, one must use `make-uri`. Otherwise, it's impossible to distinguish nil, empty string and "/".

The test that checks if the parser works properly comes below with `*test-cases*`. 

I mixed parsing and initializing the objects as to follow the style that was used before. 

I suggest using `make-uri` to test `uri=` and `uri-equal`. Do you agree?

**Comment by Ambrevar** [comment]
> The test that checks if the parser works properly comes below with `*test-cases*`.

Ah yes, indeed, I had forgotten about these.

Alright, fine with me then, I trust your judgment! :)

**Comment by aadcg** [comment]
@Ambrevar ready to merge! Thanks.

**Pull request merged**